File: Quiz.py
correct_answer=0
wrong_answer=0
percentage=0
ans_cs=[]
ans_maths=[]
print("    *******************************************QUIZ*******************************************")

print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
print()
print(" Before starting please do read the following instructions:")
print("* There will be three topics for this quiz, you will have to choose one of them.") 
print("* There would be a total of 10 questions.")
print("* For each questions you would have four possible answers.")
print("* Enter the answer which you think would be right that is either a,b,c or d")
print("* Do not enter two options, it will be considerd wrong.")
print("* At the end of this quiz, your results will be displayed.")
print("* Your results would show your right answers, wrong answers and your overall percentage.")
print('')
print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
print('')
print("To start this quiz enter the following details: ")
print('')
#registration
print("Enter your name : ")
name = input()
print('') #’\n’
print("Enter your age : ")
age = int(input())
print('')
print("Enter your email ID : ")
email = input()
print()
print("Enter your contact number : ")
contact = int(input())
print()
print()
#displaying the topics for the quiz
print("******The topics for this quiz are******")
print("1. All about Computers")
print("2. Math fun")
print("****************************************")
print("Enter your choice: ")
choice = int(input())
if choice==1:
    b=["c","a","d","b","a"]
    print("Your choice is : 1. All about Computers")
    print("Lets start the game!")
    print()
    #declare an array
    print("1.How to run Java program on the command prompt?")
    print("a.javac JavaProgram")
    print("b.javac JavaProgram.java")
    print("c.java JavaProgram")
    print("d.No one")
    print("Your option is: ")
    x=input()# inputs
    print()
    if x==b[0]:
        print("hi")
        print("correct")
        correct_answer+=1
    else:
        print("Wrong answer")
        print("The right answer is: c.java JavaProgram.")
        print("Try your luck in the next question...")
        wrong_answer+=1
    print('')
    print("2.What is the use of the println method?")
    print("a.It is used to print text on the screen with the line break.")
    print("b.It is used to print text on the screen.")
    print("c.It is used to read text from keyboard.")
    print("d.It is used to read text from a file.")
    print("Your option is: ")
    y=input()             
    print()
    if (y==(b[1])):
        print("Bravo! your answer is right :-) ")
        print("You're going on the right track!!")
        correct_answer+=1
        print('')
    else:
        print("Sorry your answer is wrong :-( ")
        print("The right answer is:a.It is used to print text on the screen with the line break.")
        print("Try your luck in the next question...")
        wrong_answer+=1
        print()
    print("3.How to read a character from the keyboard?")
    print("a.char c=System.read()")
    print("b.char c=System.in.read()")
    print("c.char c=(char)System.read()")
    print("d.char c=(char)System.in.read()")
    print("Your option is: ")
    z=input()
    print()
    if (z==(b[2])):
        print("Bravo! your answer is right :-) ")
        print("You're going on the right track!!")
        correct_answer+=1
        print()
    else:
        print("Sorry your answer is wrong :-( ")
        print("The right answer is:d.char c=(char)System.in.read()")
        print("Try your luck in the next question...")
        wrong_answer+=1
        print()
    print("4.Which one is a single-line comment?")
    print("a./...")
    print("b.//...")
    print("c./*...")
    print("d./*...*/")
    print("Your option is: ")
    ab=input()
    print()
    if (ab==(b[3])):
        print("Bravo! your answer is right :-) ")
        print("You're going on the right track!!")
        correct_answer+=1
        print()
    else:
        print("Sorry your answer is wrong :-( ")
        print("The right answer is:b.//...")
        print("Try your luck in the next question...")
        wrong_answer+=1
        print()
    print("5.How do you declare an integer variable x?")
    print("a.int x")
    print("b.x as Integer")
    print("c.Int[] x")
    print("d.No one is correct.")
    print("Your option is: ")
    ac=input()
    print()
    if (ac==(b[4])):
        print("Bravo! your answer is right :-) ")
        print("You're going on the right track!!")
        correct_answer+=1
        print()
    else:
        print("Sorry your answer is wrong :-( ")
        print("The right answer is:a.int x")
        print("Try your luck in the next question...")
        wrong_answer+=1
        print()
    #find the average percentage of the candidate
    percentage = correct_answer*20
    #function to display topic 2
if choice==2:
        d=["b","a","c","b","c"]
        print("Your choice is : 2. Math")
        print("Lets start the game!")
        print() 

        #display the question and options for topic 2
        print("1.The average of first 50 natural numbers. ")
        print("a.25.30")
        print("b.25.5")
        print("c.25.00")
        print("d.12.25")
        print("Your option is: ")
        ad=input()
        print() 
        if (ad==(d[0])):
            print("Bravo! your answer is right :-) ")
            print("You're going on the right track!!")
            correct_answer+=1
            print()
        else:
            print("Sorry your answer is wrong :-( ")
            print("The right answer is:b.25.5")
            print("Try your luck in the next question...")
            wrong_answer+=1
            print()
        print("2.A fraction which bears the same ratio to 1/27 as 3/11 bear to 5/9 is equal to?")
        print("a.1/55")
        print("b.55")
        print("c.3/11")
        print("d.1/11")
        print("Your option is: ")
        ae=input()
        print() 
        if (ae==(d[1])):
            print("Bravo! your answer is right :-) ")
            print("You're going on the right track!!")
            correct_answer+=1
            print()
        else:
            print("Sorry your anwser is wrong :-( ")
            print("The right answer is:a.1/55")
            print("Try your luck in the next question...")
            wrong_answer+=1
            print()
        print("3.The number of 3-digit numbers divisible by 6 ")
        print("a.149")
        print("b.166")
        print("c.150")
        print("d.151")
        print("Your option is: ")
        af=input()
        print()
        if (af==(d[2])):
            print("Bravo! your answer is right :-) ")
            print("You're going on the right track!!")
            correct_answer+=1
            print()
        else:
            print("Sorry your answer is wrong :-( ")
            print("The right answer is:c.150")
            print("Try your luck in the next question...")
            wrong_answer+=1
            print()

        print("4.What is 1004 divided by 2?")
        print("a.52")
        print("b.502")
        print("c.520")
        print("d.5002")
        print("Your option is: ")
        ag=input()
        print() 
        if (ag==(d[3])):
            print("Bravo! your answer is right :-) ")
            print("You're going on the right track!!")
            correct_answer+=1
            print()
        else:
            print("Sorry your answer is wrong :-( ")
            print("The right answer is:b.502")
            print("Try your luck in the next question...")
            wrong_answer+=1
            print()

        print("5.A clock strikes once at 1 o’clock, twice at 2 o’clock, thrice at 3 o’clock and so on. How many times will it strike in 24 hours?")
        print("a.78")
        print("b.136")
        print("c.156")
        print("d.196")
        print("Your option is: ")
        ah=input()
        print() 
        if (ah==(d[4])):
            print("Bravo! your answer is right :-) ")
            print("You're going on the right track!!")
            correct_answer+=1
            print()
        else:
            print("Sorry your answer is wrong :-( ")
            print("The right answer is:c.156")
            print("Try your luck in the next question...")
            wrong_answer+=1
            print()
        #find the average percentage of the candidate
        percentage = correct_answer*20

    #function display the results
else:
    print("Invalid input")

print("^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^")
print()
print()
print("                                YOUR RESULTS")
print()
print("The number of correct answers is ", correct_answer ,"/5")
print("The number of wrong answers is ", wrong_answer ,"/5")
print("Your average percentage is ",percentage ,"%")
print()
print()
print("^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^")
   
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
print()
#function to display the certificate   void certificate()
if(percentage>=80):
        
        if(choice==1):
                print("                                    CERTIFICATE OF DISTINCTION")
                print("This is to certify that ",name," has completed the online quiz on topic 1 'All About Computers' with ",percentage,"%")
        elif(choice==2):
                print("                                    CERTIFICATE OF DISTINCTION")
                print("This is to certify that ",name," has completed the online quiz on topic 2 'Math' with ",percentage,"%")
elif(percentage<80 and percentage>=50):
        if(choice==1):
                print("                                       CERTIFICATE OF MERIT")
                print("This is to certify that ",name," has completed the online quiz on topic 1 'All About Computers' with ",percentage,"%")
        elif(choice==2):
                print("                                       CERTIFICATE OF MERIT")
                print("This is to certify that ",name," has completed the online quiz on topic 2 'Math' with ",percentage,"%")
else:
        if(choice==1):
                print("                                     CERTIFICATE FOR PARTICIPATION")
                print("This is to certify that ",name," has completed the online quiz on topic 1 'All About Computers' with ",percentage,"%")
        elif(choice==2):
                print("                                     CERTIFICATE FOR PARTICIPATION")
                print("This is to certify that ",name," has completed the online quiz on topic 2 'Math' with ",percentage,"%")

File: Quiz.java
import java.util.*;
public class Project
{
    //declaration and initialization
    int correct_answer=0;
    int wrong_answer=0;
    double percentage;
    static String name;
    static int choice;

    //main function
    public static void main()
    {

        //declaration
        Scanner sc = new Scanner(System.in);

        System.out.println("    *******************************************QUIZ*******************************************");

        System.out.println("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
        System.out.println();
        System.out.println(" Before starting please do read the following instructions:");
        System.out.println("* There will be three topics for this quiz, you will have to choose one of them."); ;
        System.out.println("* There would be a total of 10 questions.");
        System.out.println("* For each questions you would have four possible answers.");
        System.out.println("* Enter the answer which you think would be right that is either a,b,c or d");
        System.out.println("* Do not enter two options, it will be considerd wrong.");
        System.out.println("* At the end of this quiz, your results will be displayed.");
        System.out.println("* Your results would show your right answers, wrong answers and your overall percentage.");
        System.out.println();
        System.out.println("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
        System.out.println();

        System.out.println("To start this quiz enter the following details: ");
        System.out.println();

        //registration
        System.out.println("Enter your name : ");
        name = sc.nextLine();
        System.out.println(); 

        System.out.println("Enter your age : ");
        int age = sc.nextInt();
        System.out.println();
        sc.nextLine();

        System.out.println("Enter your email ID : ");
        String email = sc.next();
        System.out.println();
        sc.nextLine();

        System.out.println("Enter your contact number : ");
        long contact = sc.nextLong();
        System.out.println();
        System.out.println();

        //displaying the topics for the quiz
        System.out.println("******The topics for this quiz are******");
        System.out.println("1. All about Computers");
        System.out.println("2. Math fun");
        System.out.println("****************************************");

        System.out.println("Enter your choice: ");
        choice = sc.nextInt();  

        //starting swith
        switch(choice)
        {
            case 1: 
            Project p1 = new Project();
            p1.comp();
            p1.results();
            p1.screen();
            p1.certificate();

            break;

            case 2:
            Project p2 = new Project();
            p2.math();
            p2.results();
            p2.screen();
            p2.certificate();
            break;

            default:
            System.out.println("Invalid input");

        }
    }

    //function to display topic 1
    void comp()
    {
        //declaration
        Scanner sc=new Scanner(System.in);

        //correct options
        String b[]= new String[10];
        b[0]="c";
        b[1]="a";
        b[2]="d";
        b[3]="b";
        b[4]="a";
        b[5]="c";
        b[6]="b";
        b[7]="b";
        b[8]="d";
        b[9]="c";

        System.out.println("Your choice is : 1. All about Computers");
        System.out.println("Lets start the game!");
        System.out.println();

        //declare an array
        String a[] = new String[10];

        //display the question and options for topic 1
        System.out.println("1.How to run Java program on the command prompt?");
        System.out.println("a.javac JavaProgram");
        System.out.println("b.javac JavaProgram.java");
        System.out.println("c.java JavaProgram");
        System.out.println("d.No one");
        System.out.print("Your option is: ");
        a[0]=sc.next();
        System.out.println();
        if (a[0].equals(b[0]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your answer is wrong :-( ");
            System.out.println("The right answer is: c.java JavaProgram.");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }

        System.out.println("2.What is the use of the println method?");
        System.out.println("a.It is used to print text on the screen with the line break.");
        System.out.println( "b.It is used to print text on the screen.");
        System.out.println("c.It is used to read text from keyboard.");
        System.out.println("d.It is used to read text from a file.");
        System.out.print("Your option is: ");
        a[1]=sc.next();             
        System.out.println();
        if (a[1].equals(b[1]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your answer is wrong :-( ");
            System.out.println("The right answer is:a.It is used to print text on the screen with the line break.");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }

        System.out.println("3.How to read a character from the keyboard?");
        System.out.println("a.char c=System.read()");
        System.out.println("b.char c=System.in.read()");
        System.out.println("c.char c=(char)System.read()");
        System.out.println("d.char c=(char)System.in.read()");
        System.out.print("Your option is: ");
        a[2]=sc.next();
        System.out.println();
        if (a[2].equals(b[2]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your answer is wrong :-( ");
            System.out.println("The right answer is:d.char c=(char)System.in.read()");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }

        System.out.println("4.Which one is a single-line comment?");
        System.out.println("a./...");
        System.out.println("b.//...");
        System.out.println("c./*...");
        System.out.println("d./*...*/");
        System.out.print("Your option is: ");
        a[3]=sc.next();
        System.out.println();
        if (a[3].equals(b[3]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your answer is wrong :-( ");
            System.out.println("The right answer is:b.//...");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }

        System.out.println("5.How do you declare an integer variable x?");
        System.out.println("a.int x");
        System.out.println("b.x as Integer");
        System.out.println("c.Int[] x");
        System.out.println("d.No one is correct.");
        System.out.print("Your option is: ");
        a[4]=sc.next();
        System.out.println();
        if (a[4].equals(b[4]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your answer is wrong :-( ");
            System.out.println("The right answer is:a.int x");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }
        //find the average percentage of the candidate
        percentage = correct_answer*10;

    }

    //function to display topic 2
    void math()
    {
        //declaration
        Scanner sc=new Scanner(System.in);

        //correct options
        String d[]= new String[10];
        d[0]="b";
        d[1]="a";
        d[2]="c";
        d[3]="b";
        d[4]="c";
        d[5]="b";
        d[6]="b";
        d[7]="a";
        d[8]="d";
        d[9]="d";  

        System.out.println("Your choice is : 2. Math");
        System.out.println("Lets start the game!");
        System.out.println(); 

        //declare an array
        String c[] = new String[10];

        //display the question and options for topic 2
        System.out.println("1.The average of first 50 natural numbers. ");
        System.out.println("a.25.30");
        System.out.println("b.25.5");
        System.out.println("c.25.00");
        System.out.println("d.12.25");
        System.out.print("Your option is: ");
        c[0]=sc.next();
        System.out.println(); 
        if (c[0].equals(d[0]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your answer is wrong :-( ");
            System.out.println("The right answer is:b.25.5");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }

        System.out.println("2.A fraction which bears the same ratio to 1/27 as 3/11 bear to 5/9 is equal to?");
        System.out.println("a.1/55");
        System.out.println("b.55");
        System.out.println("c.3/11");
        System.out.println("d.1/11");
        System.out.print("Your option is: ");
        c[1]=sc.next();
        System.out.println(); 
        if (c[1].equals(d[1]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your anwser is wrong :-( ");
            System.out.println("The right answer is:a.1/55");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }

        System.out.println("3.The number of 3-digit numbers divisible by 6 ");
        System.out.println("a.149");
        System.out.println("b.166");
        System.out.println("c.150");
        System.out.println("d.151");
        System.out.print("Your option is: ");
        c[2]=sc.next();
        System.out.println();
        if (c[2].equals(d[2]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your answer is wrong :-( ");
            System.out.println("The right answer is:c.150");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }

        System.out.println("4.What is 1004 divided by 2?");
        System.out.println("a.52");
        System.out.println("b.502");
        System.out.println("c.520");
        System.out.println("d.5002");
        System.out.print("Your option is: ");
        c[3]=sc.next();
        System.out.println(); 
        if (c[3].equals(d[3]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your answer is wrong :-( ");
            System.out.println("The right answer is:b.502");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }

        System.out.println("5.A clock strikes once at 1 o’clock, twice at 2 o’clock, thrice at 3 o’clock and so on. How many times will it strike in 24 hours?");
        System.out.println("a.78");
        System.out.println("b.136");
        System.out.println("c.156");
        System.out.println("d.196");
        System.out.print("Your option is: ");
        c[4]=sc.next();
        System.out.println(); 
        if (c[4].equals(d[4]))
        {
            System.out.println("Bravo! your answer is right :-) ");
            System.out.println("You're going on the right track!!");
            correct_answer++;
            System.out.println();
        }
        else
        {
            System.out.println("Sorry your answer is wrong :-( ");
            System.out.println("The right answer is:c.156");
            System.out.println("Try your luck in the next question...");
            wrong_answer++;
            System.out.println();
        }
        //find the average percentage of the candidate
        percentage = correct_answer*10;

    }
    //function display the results
    void results()
    {
        System.out.println("^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
        System.out.println();
        System.out.println();
        System.out.println("                                YOUR RESULTS");
        System.out.println();
        System.out.println("The number of correct answers is "+correct_answer+"/10");
        System.out.println("The number of wrong answers is "+wrong_answer+"/10");
        System.out.println("Your average percentage is "+percentage+"%");
        System.out.println();
        System.out.println();
        System.out.println("^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^");


    }

    //function to create an illusion 
    void screen()
    {
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
        System.out.println();
    }

    //function to display the certificate
    void certificate()
    {
        if(percentage>=80)
        {
            if(choice==1)
            {
                System.out.println("                                    CERTIFICATE OF DISTINCTION");
                System.out.println("This is to certify that "+name+" has completed the online quiz on topic 1 'All About Computers' with "+percentage+"%");
            }
            else if(choice==2)
            {
                System.out.println("                                    CERTIFICATE OF DISTINCTION");
                System.out.println("This is to certify that "+name+" has completed the online quiz on topic 2 'Math' with "+percentage+"%");
            }
        }
        else if(percentage<80 && percentage>=50)
        {
            if(choice==1)
            {
                System.out.println("                                       CERTIFICATE OF MERIT");
                System.out.println("This is to certify that "+name+" has completed the online quiz on topic 1 'All About Computers' with "+percentage+"%");
            }
            else if(choice==2)
            {
                System.out.println("                                       CERTIFICATE OF MERIT");
                System.out.println("This is to certify that "+name+" has completed the online quiz on topic 2 'Math' with "+percentage+"%");
            }
        }
        else
        {
            if(choice==1)
            {
                System.out.println("                                     CERTIFICATE FOR PARTICIPATION");
                System.out.println("This is to certify that "+name+" has completed the online quiz on topic 1 'All About Computers' with "+percentage+"%");
            }
            else if(choice==2)
            {
                System.out.println("                                     CERTIFICATE FOR PARTICIPATION");
                System.out.println("This is to certify that "+name+" has completed the online quiz on topic 2 'Math' with "+percentage+"%");
            }
        }
    }
}
